client/components/domains/search-filters/tld-filter-bar.jsx

```diff
@@ -0,0 +1,207 @@
+		onReset: PropTypes.func.isRequired,
+		onSubmit: PropTypes.func.isRequired,
+		recordTldFilterSelected: PropTypes.func.isRequired,
+		showPlaceholder: PropTypes.bool.isRequired,
+	};
+
+	static defaultProps = {
+		numberOfTldsShown: 8,
+	};
+
+	state = {
+		showPopover: false,
+	};
+
+	bindButton = button => ( this.button = button );
+
+	handleButtonClick = event => {
+		const isCurrentlySelected = event.currentTarget.dataset.selected === 'true';
+		const newTld = event.currentTarget.value;
+
+		this.props.recordTldFilterSelected(
+			newTld,
+			event.currentTarget.dataset.index,
+			! isCurrentlySelected
+		);
+
+		const tldSet = new Set( [ ...this.props.filters.tlds, newTld ] );
+		if ( isCurrentlySelected ) {
+			tldSet.delete( newTld );
+		}
+
+		const tlds = [ ...tldSet ];
+		this.props.onChange( { tlds }, { shouldSubmit: true } );
+	};
+	handleFiltersReset = () => {
+		this.togglePopover();
+		this.props.onReset( 'tlds' );
+	};
+	handleFiltersSubmit = () => {
+		this.togglePopover();
+		this.props.onSubmit();
+	};
+	handleTokenChange = newTlds => {
+		const tlds = [
+			...new Set( [
+				...this.getSelectedSuggestedTlds(),
+				...newTlds.filter( tld => includes( this.props.availableTlds, tld ) ),
+			] ),
+		];
+
+		this.props.onChange( { tlds } );
+	};
+
+	togglePopover = () => {
+		this.setState( {
+			showPopover: ! this.state.showPopover,
+		} );
+	};
+
+	getAvailableSuggestedTlds() {
+		return this.props.availableTlds.slice( 0, this.props.numberOfTldsShown );
+	}
+
+	getAvailableRegularTlds() {
+		return this.props.availableTlds.slice( this.props.numberOfTldsShown );
+	}
+
+	getSelectedSuggestedTlds() {
+		return this.props.filters.tlds.filter( tld =>
+			includes( this.getAvailableSuggestedTlds(), tld )
+		);
+	}
+
+	getSelectedRegularTlds() {
+		return this.props.filters.tlds.filter(
+			tld => ! includes( this.getAvailableSuggestedTlds(), tld )
+		);
+	}
+
+	render() {
+		if ( this.props.showPlaceholder ) {
+			return this.renderPlaceholder();
+		}
+
+		return (
+			<CompactCard className="search-filters__buttons">
+				{ this.renderSuggestedButtons() }
+				{ this.renderPopoverButton() }
+				{ this.state.showPopover && this.renderPopover() }
+			</CompactCard>
+		);
+	}
+
+	renderSuggestedButtons() {
+		const { lastFilters: { tlds: selectedTlds } } = this.props;
+		return this.getAvailableSuggestedTlds().map( ( tld, index ) => (
+			<Button
+				className={ classNames( { 'is-active': includes( selectedTlds, tld ) } ) }
+				data-selected={ includes( selectedTlds, tld ) }
+				data-index={ index }
+				key={ tld }
+				onClick={ this.handleButtonClick }
+				value={ tld }
+			>
+				.{ tld }
+			</Button>
+		) );
+	}
+
+	renderPopoverButton() {
+		return (
+			<Button
+				className={ classNames( { 'is-active': this.getSelectedRegularTlds().length > 0 } ) }
+				onClick={ this.togglePopover }
+				ref={ this.bindButton }
+				key="popover-button"
+			>
+				{ this.props.translate( 'More Extensions', {
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Extensions', { context: 'Refers to top level domain name extension, such as ".com"'} )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.